### PR TITLE
Feature/encounter be hookup

### LINKED ISSUE #19: As a user I should be able to click on a area of the map, and have a random encounter with a monster or a NPC (non player character)
Enemies

- [x] Create Giant Crab
- [x] Create Mountain Troll
- [x] Create Lizard Man
- [x] Create Cave Spider
- [x] Create Goblin
- [x] Create Evil Elf

Friendly
- [x] Create Jeff
- [x] Create Farmer
- [x] Create Circus Actor

### PR DESCRIPTION
**Pull request by rubyr**
## Description

This branch hooks the back end up to the front end, allowing the user to click on a map location and get into an encounter with a random NPC from the backend, based on the location they chose. 

Fixes #19

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Manual testing for now, but I will be working on a test branch soon 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
    - Error when loading into an encounter, not sure how to fix
- [ ] I have added tests that prove my fix is effective or that my feature works
    - testing TBD
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/Description/Description.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import { usePlayerState } from "../playerState";
+import { useEncounterState } from "../encounterState";
 
 const Description = () => {
   const player = usePlayerState();
+  const { npc } = useEncounterState();
   // starting string, can be changed according to player's last location
-  let description = "You are in a grassy field. ";
+  let description =
+    npc !== null ? `You see a ${npc.name}. ` : "You are in a grassy field. ";
```
Comment (Maxwell-Baird) [inline src/Description/Description.js:11]: Two questions, first should a npc have a description that we can insert here or is this going to be wear insert the first part dialogue path?

File: src/encounterState/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  if (dispatch === undefined) {
+    throw new Error("useDispatch used outside of provider");
+  }
+
+  const getNewEncounter = useCallback(
+    async (area) => {
+      dispatch({ type: EncounterActions.NEW_ENCOUNTER_LOAD });
+
+      const allNpcs = await (
+        await fetch("http://localhost:8000/api/v1/npcs")
```
Comment (Maxwell-Baird) [inline src/encounterState/index.js:44]: We probably want to change this to be an environment variable so when we deploy on production we can change the url

File: src/Description/Description.js (hunk 1 above)
Comment (rubyr) [reply src/Description/Description.js:11]: I think having an NPC description would be good! It'd certainly make for more dynamic dialogue than this.

File: src/encounterState/index.js (hunk 2 above)
Comment (rubyr) [reply src/encounterState/index.js:44]: Of course, this was just to get things running. Will look into environment variables shortly

File: src/Encounters/Encounters.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+        <>
+          <h1>{npc.name}</h1>
+          <h3>
+            {npc.health} / {npc.maxHealth} hp
+          </h3>
+          <h3>ATK: {npc.attack}</h3>
+          <h3>DEF: {npc.defense}</h3>
+        </>
+      ) : (
+        "Loading..."
```
Comment (KirbyDD) [inline src/Encounters/Encounters.js:23]: Are you guys thinking about using thunks for things like this?

**Pull request merged**